src/versionLoader.ts

```diff
@@ -0,0 +1,35 @@
+import { MODE, URLS } from './constants'
+import { Mode } from './types'
+import { loadModule } from './utils'
+
+let loadedVersion: string = null
+
+export function RevolutPaymentsVersionLoader(
+  mode: Mode = RevolutPaymentsVersionLoader.mode
+): Promise<string> {
+  if (typeof loadedVersion === 'string') {
+    return Promise.resolve(loadedVersion)
+  }
+
+  return loadModule({
+    src: URLS[mode].version,
+    id: 'revolut-pay-version',
+    name: 'RevolutPayVersion',
+  })
+    .then(() => {
+      loadedVersion =
+        '__REV_PAY_VERSION__' in window &&
+        typeof __REV_PAY_VERSION__ === 'string'
+          ? __REV_PAY_VERSION__
+          : ''
+      delete window.__REV_PAY_VERSION__
+
+      return loadedVersion
+    })
+    .catch(() => {
+      loadedVersion = ''
```

Comment:
do you intentionally assign empty string here?

Reply:
Yes. What else could we have here?
Version file is not mission critical, so if it fails, so be it.